### PR TITLE
Adding support for multiple results on query

### PR DESCRIPTION
**Pull request by alcasim**
## Pull Request Checklist

**Is this in reference to an existing issue?**

#### General

- [x] Update Changelog following the conventions laid out on [Keep A Changelog](http://keepachangelog.com/)

- [x] Update README with any necessary configuration snippets

- [x] Binstubs are created if needed

- [x] RuboCop passes

- [x] Existing tests pass

### CODE REVIEW

File: bin/check-influxdb-query.rb [hunk 1]
```diff
@@ -179,19 +185,45 @@ def run
...
-
       calc = Dentaku::Calculator.new
-      if config[:critical] && calc.evaluate(config[:critical], value: value)
-        critical "Value '#{value}' matched '#{config[:critical]}' for query '#{query_name}'"
-      elsif config[:warning] && calc.evaluate(config[:warning], value: value)
-        warning "Value '#{value}' matched '#{config[:warning]}' for query '#{query_name}'"
+      if config[:mode] == 'any' && json_path.on(value).length >= 1
+        json_path.on(value).each do |hashval|
+          if config[:critical] && calc.evaluate(config[:critical], value: hashval)
+            critical "Value '#{value}' matched '#{config[:critical]}' for query '#{query_name}'"
```
Comment (majormoses) [inline bin/check-influxdb-query.rb:192]: won't the single quotes prevent string interpolation here?

File: bin/check-influxdb-query.rb [hunk 2]
```diff
@@ -179,19 +185,45 @@ def run
...
+          elsif config[:warning] && calc.evaluate(config[:warning], value: hashval)
+            warning "Value '#{value}' matched '#{config[:warning]}' for query '#{query_name}'"
```
Comment (majormoses) [inline bin/check-influxdb-query.rb:194]: won't the single quotes prevent string interpolation here?

File: bin/check-influxdb-query.rb [hunk 3]
```diff
@@ -179,19 +185,45 @@ def run
...
+          end
+        end
+        ok 'All values OK!'
       else
-        ok "Value '#{value}' ok for query '#{query_name}'"
+        value = get_single_value(json_path, value)
+        if config[:critical] && calc.evaluate(config[:critical], value: value)
+          critical "Value '#{value}' matched '#{config[:critical]}' for query '#{query_name}'"
```
Comment (majormoses) [inline bin/check-influxdb-query.rb:201]: won't the single quotes prevent string interpolation here?

File: bin/check-influxdb-query.rb [hunk 4]
```diff
@@ -179,19 +185,45 @@ def run
...
+        elsif config[:warning] && calc.evaluate(config[:warning], value: value)
+          warning "Value '#{value}' matched '#{config[:warning]}' for query '#{query_name}'"
+        else
+          ok "Value '#{value}' ok for query '#{query_name}'"
```
Comment (majormoses) [inline bin/check-influxdb-query.rb:205]: won't the single quotes prevent string interpolation here?

File: bin/check-influxdb-query.rb (hunk 1 above)
Comment (luisdavim) [reply bin/check-influxdb-query.rb:192]: I don't think so:
```
$ pry
[1] pry(main)> a='ff'
=> "ff"
[2] pry(main)> b='cccc'
=> "cccc"
[3] pry(main)> p "a is '#{a}' and b is '#{b}'"
"a is 'ff' and b is 'cccc'"
=> "a is 'ff' and b is 'cccc'"
```

File: bin/check-influxdb-query.rb (hunk 2 above)
Comment (luisdavim) [reply bin/check-influxdb-query.rb:194]: They are not necessary but they won't prevent interpolation.

File: bin/check-influxdb-query.rb (hunk 3 above)
Comment (luisdavim) [reply bin/check-influxdb-query.rb:201]: They are not necessary but they won't prevent interpolation.

File: bin/check-influxdb-query.rb (hunk 4 above)
Comment (luisdavim) [reply bin/check-influxdb-query.rb:205]: They are not necessary but they won't prevent interpolation.

File: bin/check-influxdb-query.rb (hunk 1 above)
Comment (majormoses) [reply bin/check-influxdb-query.rb:192]: Interesting, I validated this behavior.

**Review by majormoses — APPROVED** [review]
LGTM

**Comment by majormoses** [comment]
@luisdavim please ping me if I do not release within 24 hours

**Comment by majormoses** [comment]
released: https://rubygems.org/gems/sensu-plugins-influxdb/versions/1.2.0

**Comment by luisdavim** [comment]
Thanks @majormoses

**Pull request merged**